replication_types.go

```diff
@@ -114,7 +114,7 @@ type ProtectionPolicy struct {
 }
 
 func (policy *ProtectionPolicy) Fields() []string {
-	return []string{"*"}
+	return []string{"id", "name", "replication_rules", "description", "snapshot_rules", "type", "is_read_only"}
```

Comment:
"*" was added to maintain backward compatibility. Please check the powerstore version in which "description", "snapshot_rules", "type" & "is_read_only" were added.

Reply:
"description", "snapshot_rules", "type" & "is_read_only" were added recently through https://github.com/dell/gopowerstore/commit/870b2aa283a3ff75c77e3596665e379bed5f5026 and https://github.com/dell/gopowerstore/commit/fd8afd60ae81f3cebe97a50ce8bcbd60b28f6ad6

Actually, with "*" we are not getting "replication_rules" and "snapshot_rules" in the response.
snippet from log:

```
2023-02-02T10:42:17.331-0500 [DEBUG] provider.terraform-provider-powerstore: 2023/02/02 10:42:17 REQUEST: GET /api/rest/policy/cb3af189-4f54-40c6-b56e-03adb06a0ff1?select=%2A HTTP/1.1 Host: 10.230.24.94 Authorization: ******
2023-02-02T10:42:17.331-0500 [DEBUG] provider.terraform-provider-powerstore: 2023/02/02 10:42:17 acquire a lock
2023-02-02T10:42:17.370-0500 [DEBUG] provider.terraform-provider-powerstore: 2023/02/02 10:42:17 RESPONSE: HTTP/1.1 200 OK Content-Length: 290 Cache-Control: no-cache Cache-Control: no-store Cache-Control: must-revalidate Cache-Control: max-age=0 Content-Language: en-US Content-Type: application/json Dell-Emc-Token: ****** Expires: -1 Set-Cookie: auth_cookie=******; Path=/; Secure; HTTPOnly X-Content-Type-Options: nosniff  {"id":"cb3af189-4f54-40c6-b56e-03adb06a0ff1","name":"test_protection_policy","description":"Creating Protection Policy","type":"Protection","managed_by":"User","managed_by_id":null,"is_read_only":false,"is_replica":false,"type_l10n":"Protection","managed_by_l10n":"Managed by the end user"}
2023-02-02T10:42:17.371-0500 [DEBUG] provider.terraform-provider-powerstore: 2023/02/02 10:42:17 release a lock
2023-02-02T10:42:17.371-0500 [DEBUG] provider.terraform-provider-powerstore: 2023/02/02 10:42:17 Done with Create
```

Reply:
Please remove is_read_only, it was added in 3.0.0.